consensus/istanbul/core/commit.go

```diff
@@ -84,38 +86,62 @@ func (c *core) handleCommit(msg *istanbul.Message) error {
 		return err
 	}
 
-	if err := c.verifyCommit(commit); err != nil {
-		return err
-	}
-
-	_, validator := c.valSet.GetByAddress(msg.Address)
-	if validator == nil {
-		return errInvalidValidatorAddress
-	}
+	msgSequencePlusOne := big.NewInt(0).Add(commit.View.Sequence, common.Big1)
+	// if the received view's sequence +1 equals the current view, then the
+	// received view corresponds to the parent block
+	if c.currentView().Sequence.Cmp(msgSequencePlusOne) == 0 {
+		// Retrieve the validator set for the previous proposal (which should
+		// match the one broadcast)
+		parentValset := c.backend.ParentValidators(c.current.Proposal())
```

Comment:
Is this an off-by-one error? IIUC, `ParentValidators` will give us the validator set that needs to sign off on the block we pass in, in this case, `c.current.Proposal()` which is the block we're currently trying to finalize. 

I think instead we want to pass `c.backend.LastProposal()`, but please double check me here

Reply:
Yeah, this and the other off-by-one eliminated the errors in the backend which were appearing some times

Reply:
For posterity, we agreed that it was